Handle SkColor creation from rgba tuple

louix opened the pull request:
`Skia.Color` accepts `number[]`, but the native constructor doesn't actually account for it.

In [JsiSkColor::createCtor][0], if `argument[0].isObject()` (true for both `Array` and `Float32Array`), it's returned as is as the color value.

When the `jsi::Value` is made into to an `SkColor` in [JsiSkColor::FromValue][1], the `.buffer` property (which is of course is not there on an array -- thus undefined) is attempted to be accessed.

I've added support here.

Orthogonally, have you considered using `[r: number, g: number, b: number, a: number]` instead of `number[]` for extra type saftey?

Resolves #2113

[0]: https://github.com/Shopify/react-native-skia/blob/71161bdeb579a10e7988d39f0f62bd4582c34446/package/cpp/api/JsiSkColor.h#L66
[1]: https://github.com/Shopify/react-native-skia/blob/71161bdeb579a10e7988d39f0f62bd4582c34446/

louix committed:
Handle SkColor creation from rgba tuple

wcandillon committed:
Merge branch 'main' into main

wcandillon committed:
Add tests

wcandillon commented:
Thanks a lot I'm excited to sanitize the situation there. I added some tests for it at https://github.com/louix/react-native-skia/pull/1 but it seems to fail on Android.
I will look into it.

louix committed:
Merge pull request #1 from Shopify/louix/main

Add test

louix committed:
Fix Float32Array check

louix committed:
Add offscreen canvas test

louix committed:
Add exhaustive web case (to match C++)

wcandillon committed:
Merge branch 'main' into main

louix commented:
You were correct about 0-1 vs 0-255!
I changed the JSI side slightly to mimic the web one (Array<number> become Float32Array), and added the offscreen canvas test.

The native side was just returning undefined if it wasn't sure how to handle the value, so you'd get an error where you tried to use the SkColor instead of when you created it.
I added an throw at creation time (and an exhaustive one to the web side to match).

wcandillon commented on package/cpp/api/JsiSkColor.h:
```diff
@@ -76,9 +73,24 @@ class JsiSkColor : public RNJsi::JsiHostObject {
+        }
       }
-      return jsi::Value::undefined();
+      throw jsi::JSError(runtime,
```
yes that's fair

wcandillon commented:
but either way the tests you've written will be very useful.
